botorch/optim/initializers.py

```diff
@@ -226,6 +228,131 @@ def gen_one_shot_kg_initial_conditions(
+            `frac_random` (the fraction of fully random fantasy points),
+            `num_inner_restarts` and `raw_inner_samples` (the number of random
+            restarts and raw samples for solving the posterior objective
+            maximization problem, respectively) and `eta` (temperature parameter
+            for sampling heuristic from posterior objective maximizers).
+
+    Returns:
+        A `num_restarts x batch_shape x q x d` tensor that can be used as initial
+        conditions for `optimize_acqf()`. Here `batch_shape` is the
+        `_input_batch_shape` of value function model.
+
+    Example:
+        >>> fant_X = torch.rand(5, 1, 2)
+        >>> fantasy_model = model.fantasize(fant_X, SobolQMCNormalSampler(16))
+        >>> value_function = PosteriorMean(fantasy_model)
+        >>> bounds = torch.tensor([[0., 0.], [1., 1.]])
+        >>> Xinit = gen_value_function_initial_conditions(
+        >>>     value_function, bounds, num_restarts=10, raw_samples=512,
+        >>>     options={"frac_random": 0.25},
+        >>> )
+    """
+    options = options or {}
+    seed: Optional[int] = options.get("seed")
+    frac_random: float = options.get("frac_random", 0.6)
+    if not 0 < frac_random < 1:
+        raise ValueError(
+            f"frac_random must take on values in (0,1). Value: {frac_random}"
+        )
+
+    # compute maximizer of the current value function
+    value_function = _get_value_function(
+        model=current_model,
+        objective=acq_function.objective,
+        sampler=acq_function.sampler
+        if isinstance(acq_function, MCAcquisitionFunction)
+        else None,
+    )
+    from botorch.optim.optimize import optimize_acqf
+
+    fantasy_cands, fantasy_vals = optimize_acqf(
+        acq_function=value_function,
+        bounds=bounds,
+        q=1,
+        num_restarts=options.get("num_inner_restarts", 20),
+        raw_samples=options.get("raw_inner_samples", 1024),
+        return_best_only=False,
+        options=options,
+    )
+
+    batch_shape = acq_function.model._input_batch_shape
+    # sampling from the optimizers
+    n_value = int((1 - frac_random) * raw_samples)  # number of non-random ICs
+    if n_value > 0:
+        eta = options.get("eta", 2.0)
+        weights = torch.exp(eta * standardize(fantasy_vals))
+        idx = batched_multinomial(
+            weights=weights.expand(*batch_shape, -1),
+            num_samples=n_value,
+            replacement=True,
+        ).permute(-1, *range(len(batch_shape)))
+        resampled = fantasy_cands[idx]
+    else:
+        resampled = torch.empty(0, *batch_shape, 1, bounds.shape[-1])
+    # add qMC samples
+    if raw_samples > n_value:
+        randomized = draw_sobol_samples(
+            bounds=bounds,
+            n=raw_samples - n_value,
+            q=1,
+            batch_shape=batch_shape,
+            seed=seed,
+        )
+    else:
+        randomized = torch.empty(0, *batch_shape, 1, bounds.shape[-1])
```

Comment:
looks like this isn't covered by any test - mind adding one? In fact, looks like this will break with non-default dtype..

Reply:
Actually, that line is unreachable. Due to `if not 0 < frac_random < 1: raise ValueError` and `n_value = int((1 - frac_random) * raw_samples)`, `raw_samples` will be at least 1 greater than `n_value`. 

I removed that if/else and fixed the dtype issue. This made one of the test cases redundant, so I removed that too